### DISCUSSION TITLE
Page is requesting *.js and styles from root url and not the repo specific url

### DISCUSSION THREAD

**Discussion by jupmorenor** [community/community#75680] [Other Feature Feedback, Questions, & Ideas] [answered]
Hey!
I've finally managed to configure a Github Action to build my Angular (v16) web app and it is working very well, it is commiting the files in the gh-pages branch and the files are being deployed correctly.

My issue is as follows: When I open the web page, the index.html file is loaded correctly, but when it tries to load scripts and styles it gets a 404 error for these files. Checking on devtools I can see the request is done for the main url and not for the repo specific url.

So, for example: 
- request url is https://username.github.io/main.24eb1101ed9fa6
- but it must be https://username.github.io/repo-name/main.24eb1101ed9fa6

I also have a web page deployed for the main repo username.github.io. Someone said in another discussion that I may move the files to a folder in the main repo but I don't think it's the real solution to my issue.

I hope you can give me some advice. :smi

**UPDATE**
I've tried this: https://github.com/orgs/community/discussions/52368#discussioncomment-7037954  with no luck.

**Comment by ivnvxd** [comment, marked as answer]
Angular applications need to know the path where the application is hosted. If it's not the root path, you need to set the base href accordingly.

You can set the `baseHref` property in your `angular.json` to `"/repo-name/"`. This tells Angular the root directory of your app on the server.

Or when building your Angular app for production, you can use the `--base-href` option with the Angular CLI:
```sh
ng build --prod --base-href="/repo-name/"
```

**Reply by jupmorenor** [reply]
That's the trick!
Thank you for your answer.
I hope someone can add this to the guide of 404 troubleshooting

**Reply by ivnvxd** [reply]
I'm glad it helped 😊

**Comment by jupmorenor** [comment]
Facing the same issue with a new project so I had to search for my own post to recover the solution on that time
